### ISSUE TITLE
Black Screen on Samsung A52

### ISSUE THREAD

**Issue by dyvoker** [androidx/media#321]
### Media3 Version

Media3 1.0.0

### Devices that reproduce the issue

Samsung A52 running Android 13

### Devices that do not reproduce the issue

Pixel 5a running Android 13

### Reproducible in the demo app?

Not tested

### Reproduction steps

1. Find video with big resolution (~4k) 60fps, mp4 format, avc codec.
2. Use ProgressiveMediaSource and cache.
3. Setting of the Player will not affect decoding because only one file is available.
4. Try to play it.

### Expected result

The media plays successfully (with lags, like in version ExoPlayer = "2.17.0").

### Actual result

Black screen

### Media

https://video.wbstatic.net/video/new/12800000/12801489.mp4

### Bug Report

- [ ] You will email the zip file produced by `adb bugreport` to dev.exoplayer@gmail.com after filing this issue.

**Comment by icbaker** [comment]
Please reproduce the issue using the [demo app](https://developer.android.com/guide/topics/media/exoplayer/demo-application), take an `adb bugreport` immediately after reproducing and send it to dev.exoplayer@gmail.com with the subject `Issue #321`. Please also update this issue to indicate you’ve done this.

**Comment by icbaker** [comment]
The bug report seems to contain connection errors, like:

```
E EventLogger: internalError [eventTime=5.71, mediaPos=0.00, window=0, loadError
E EventLogger:   androidx.media3.datasource.cronet.CronetDataSource$OpenException: is: Exception in CronetUrlRequest: net::ERR_CERT_AUTHORITY_INVALID, ErrorCode=11, InternalErrorCode=-202, Retryable=false
E EventLogger:       at androidx.media3.datasource.cronet.CronetDataSource.open(CronetDataSource.java:581)
```

Which ultimately cause playback to fail before it starts:

```
EventLogger: playerFailed [eventTime=5.74, mediaPos=0.00, window=0, period=0, errorCode=ERROR_CODE_IO_NETWORK_CONNECTION_FAILED
E EventLogger:   androidx.media3.exoplayer.ExoPlaybackException: Source error
E EventLogger:       at androidx.media3.exoplayer.ExoPlayerImplInternal.handleIoException(ExoPlayerImplInternal.java:652)
E EventLogger:       at androidx.media3.exoplayer.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:624)
E EventLogger:       at android.os.Handler.dispatchMessage(Handler.java:102)
E EventLogger:       at android.os.Looper.loopOnce(Looper.java:226)
E EventLogger:       at android.os.Looper.loop(Looper.java:313)
E EventLogger:       at android.os.HandlerThread.run(HandlerThread.java:67)
E EventLogger:   Caused by: androidx.media3.datasource.cronet.CronetDataSource$OpenException: is: Exception in CronetUrlRequest: net::ERR_CERT_AUTHORITY_INVALID, ErrorCode=11, InternalErrorCode=-202, Retryable=false
E EventLogger:       at androidx.media3.datasource.cronet.CronetDataSource.open(CronetDataSource.java:581)
```

Does this mean that Cronet on the Samsung A52 is complaining that `https://storage.googleapis.com/wvmedia/cenc/h264/tears/tears.mpd` has an invalid certificate? That seems surprising...

What happens if you use a different HTTP stack by flipping this line to `false` in the demo app?
https://github.com/androidx/media/blob/62afbe87bb05890bbbb60ff4e3f0b2fd65aea433/demos/main/src/main/java/androidx/media3/demo/main/DemoUtil.java#L56

**Comment by dyvoker** [comment]
Set `USE_CRONET_FOR_NETWORKING = false;`. And it's started to work on Samsung A52.

**Comment by icbaker** [comment]
Thanks, this seems to be a device-specific issue with the cronet library. I will try and get hold of a Samsung A52 to see if I can reproduce (tried on a Samsung A51 and it didn't reproduce).

**Comment by dyvoker** [comment]
Hi, Ian! Any news? I still waiting for the fix, can't update the library because of this bug... @icbaker

**Comment by icbaker** [comment]
No update - which library are you blocked on updating? Since this seems to be a bug in cronet is that the library you're trying to update? I would recommend reporting the issue directly to them. I don't think any fix for this will end up in androidx.media3.

**Comment by dyvoker** [comment]
Thanks for the answer! I need to update androidx.media3. I updated it in my branch and the problem appears on this device. Do you know where I can create an issue for them?

**Comment by icbaker** [comment]
What version were you upgrading from/to? It looks like we updated our version of Cronet from 17.0.1 to 18.0.1 and that was included in media3 `1.0.0-alpha02`: https://github.com/androidx/media/commit/d2cc14a98bc44ce5f9fb17bd1d1b6bd909f13d19

You could try configuring gradle to use the old dependency and see if that resolves your problem (it's not an ideal permanent fix, but it further isolates the problem to the Cronet dep): https://docs.gradle.org/current/userguide/dependency_downgrade_and_exclude.html

However this might not actually affect the exact Cronet implementation you get, because that's delivered by Google Play Services. See the README for more info (and instructions for how to depend on an exact version of Cronet **not** delivered by Google Play Services): https://github.com/androidx/media/tree/release/libraries/datasource_cronet#cronet-implementations

---

Or you could use a different networking stack: https://developer.android.com/guide/topics/media/exoplayer/network-stacks

---

> Do you know where I can create an issue for them?

There's info on browsing and filing bugs at the bottom of https://developer.android.com/guide/topics/connectivity/cronet

**Comment by icbaker** [comment]
Thanks for the update - that explanation doesn't really explain the stack trace I quoted from the BR in https://github.com/androidx/media/issues/321#issuecomment-1549884591 - but glad you got things sorted! I'll close this - please open a new issue if you have further questions.

**Comment by dyvoker** [comment]
Yeah, the problem with the Cronet was that for our backend we need to specify an additional header, otherwise it doesn't allow us to load video...
